admin_solutions/authentication.adoc

```diff
@@ -5,10 +5,308 @@ toc::[]
 {product-title} supports many different authentication methods, as defined in
 xref:../install_config/configuring_authentication.adoc#install-config-configuring-authentication[Configuring Authentication]:
 
+- xref:basic-auth-remote[Basic Authentication (Remote)]
+- xref:request-header-auth[Request Header]
 - xref:keystone-auth[Keystone]
 - xref:ldap-auth[LDAP]
 - xref:github-auth[GitHub]
 
+[[basic-auth-remote]]
+== Basic Authentication (Remote)
+
+[[BasicAuthPasswordIdentityProvider]]
+
+Configuring Basic Authentication allows users to log in to {product-title} with
+credentials stored on a remote identity provider.
```

Comment:
This is not specific enough... all the auth integrations do that.

BasicAuthPasswordIdentityProvider and RequestHeaderIdentityProvider are the most generic integration mechanisms.

BasicAuthPasswordIdentityProvider is a backend integration. It makes users send their username/password to OpenShift, and OpenShift then verifies those credentials against an external server by making a server-to-server request, passing the given credentials as a basic auth header.
* This only works for username/password login mechanisms
* OpenShift must be able to make network requests to the remote auth server
* Requires users to send their credentials to OpenShift during login

RequestHeaderIdentityProvider makes users talk to OpenShift through a proxy, which is responsible for 1) authenticating the user any way it wants to, and 2) providing the user's identity to OpenShift via a request header.
* The proxy must be able to make network requests to the OpenShift server
* Unauthenticated login attempts are redirected to a configured proxy URL
* The proxy can authenticate browser clients any way it wants to, but must currently use basic auth or kerberos if it wants to work with the `oc` CLI tooling.
* Means user credentials can be handled by the proxy and never seen by OpenShift